catalog/core/catalog-core-attributeregistry/src/main/java/ddf/catalog/data/impl/AttributeRegistryImpl.java

```diff
@@ -43,27 +44,77 @@ public boolean register(final AttributeDescriptor attributeDescriptor) {
 
         final String name = attributeDescriptor.getName();
 
-        if (attributeMap.putIfAbsent(name, attributeDescriptor) == null) {
+        if (attributeMap.containsKey(name)) {
+            AttributeValueType attributeValueType = attributeMap.get(name);
+            AttributeDescriptor comparableAttributeDescriptor =
+                    attributeValueType.getAttributeDescriptor();
+            if (comparableAttributeDescriptor.equals(attributeDescriptor)) {
+                attributeValueType.incrementOccurrences();
```

Comment:
Ah yes, given that `deregister()` uses the name of the attribute, the count should be incremented either way.  Just something to note : if one were to define a previously registered attribute in JSON with same name and different multivalued and tokenized properties, it would be registered (occurrence would be incremented) with the properties of the previously registered attribute.